utils.bzl

```diff
@@ -0,0 +1,28 @@
+"""Utility macros."""
+
+def boolean_extension_constraint(name):
+    """
+    Macro for defining constraints for an extension that is optionally present.
+
+    Defines a `constraint_setting`, <name>_setting and two `constraint_values`,
+    no_<name> and <name>.
+    The setting defaults to `no_<name>` for maximum compatibility, since
```

Comment:
I don't believe this is needed.
Just create the singleton pair of constraint_setting and constraint_value.
If someone really, really needs to test for a constraint being unset, they can do a select on the constraint value vs. conditions_default.